### PR TITLE
Add related history

### PR DESCRIPTION
**Pull request by cristina0botez**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -26,26 +26,44 @@ Additional features:
...
                       an additional field called however you like and pass it on
                       to the HistoryLogging object. This field must be a dictionary
                       that contains the keys 'user_ids' and 'user_names', both values
-                      for these keys must be lists containing the appropriate information.
+                      for these keys must be lists containing the appropriate information::
+
                            ignore_history_for_users = {'user_ids': [1010101], 'user_names': ['ignore_user']}
                            history = HistoryLogging(ignore_history_for_users='ignore_history_for_users')
 
+   - Interested related fields -
```
Comment (bogdanpop) [inline README.rst:54]: Interested?

File: README.rst [hunk 2]
```diff
@@ -26,26 +26,44 @@ Additional features:
...
+                       *(added in version 1.1.0)*
+                       
+                       if you wish to add history for the objects related to a model
+                       when the model changes, you can do so by declaring a list with the names of
+                       the related fields names. This is applicable to 1-to-1, 1-to-many and
+                       many-to-many fields::
+
+                          poll = ForeignKey('Poll')
+                          ...
+                          interested_related_fields = ['poll']
```
Comment (bogdanpop) [inline README.rst:64]: How about  tracked related fields? Interested is definitely not the word you're looking for.

File: src/atris/models/abstract_historical_record.py [hunk 3]
```diff
@@ -90,9 +106,9 @@ def previous_version_by_model_and_id(self, model, object_id, history_id):
             id__lt=history_id
         )
         if main_qs.count() <= 0:
```
Comment (bogdanpop) [inline src/atris/models/abstract_historical_record.py]: I wonder if we shouldn't use `exists()` here instead? I believe django also has this, no? It should be way more efficient.

File: src/atris/models/abstract_historical_record.py [hunk 4]
```diff
@@ -90,9 +106,9 @@ def previous_version_by_model_and_id(self, model, object_id, history_id):
...
-            return None
-        result = main_qs.order_by('-history_date').first()
-
+            result = None
```
Comment (bogdanpop) [inline src/atris/models/abstract_historical_record.py]: This condition could be compacted into:

```
if exists():
   return main_qs.order_by....
return None
```

**Review by bogdanpop — COMMENTED** [review]
Uhhh, too much has changed for me to give a proper review without actually literally going through the code and using it a bit.

I left some very minor comments that are not that relevant... It's as much as I can do without going through the logic step by step, which I don't really have time for.

At a glance, it looks okay! :+1:

File: README.rst (hunk 2 above)
Comment (cristina0botez) [reply README.rst:64]: The behaviour I'm going for is the following: When an object changes, generate a history record for the object(s) that is referenced by the related field. So basically these objects are interested in any changes this object goes through. There might be a better word than interested, but tracked is not it :)

Comment (bogdanpop) [reply README.rst:64]: My gripe is that "interested" is not a word one would use on an object, an object can be OF interest, but it cannot be interested in something.

Comment (cristina0botez) [reply README.rst:64]: Sure, an object A is of interest to another object B. Which makes B interested in object A.
Say you have an episode belonging to a show. The models would look something like this:
```
class Show(Model):
  ...

class Episode(Model):
  ...
  show = ForeignKey(Show)
```
When changes are made to an episode, the show that owns the episode would want to know of those changes. So `show` as a field represents and object that is interested in the history of the owned episode.

Comment (bogdanpop) [reply README.rst:64]: I understand what it does, my issue here is with the naming, what I meant is that the word is not usable like this, but I can't say I feel strongly enough about it to insist that it be changed, nor can I think of something concise that would work, not now anyway. :man_shrugging:

**Review by mvmocanu — APPROVED** [review]

**Pull request merged**